### PR TITLE
Attempt to reduce spammy logs from external cred helpers

### PR DESCRIPTION
**Pull request by imjasonh**

### CODE REVIEW

**Comment by imjasonh** [comment]
I'm not sure why `ko` with https://github.com/google/ko/pull/581 logs `pkg/v1/google.Keychain` errors, since AFAIK `logs.Debug` is discarded by default, and I don't see where `ko` changes that, if anywhere.

I'll need to look closer, ideas welcome.

**Comment by imjasonh** [comment]
lol nevermind this doesn't work anyway

File: pkg/authn/keychain.go [hunk 1]
```diff
@@ -154,6 +155,16 @@ func NewKeychainFromHelper(h Helper) Keychain { return wrapper{h} }
 type wrapper struct{ h Helper }
 
 func (w wrapper) Resolve(r Resource) (Authenticator, error) {
+	// Route any logging from the cred helper to logs.Debug, since some of
+	// these may produce spammy logs that we don't necessarily care about.
+	stdout, stderr := os.Stdout, os.Stderr
+	os.Stdout = logs.Debug.Writer()
```
Comment (mattmoor) [inline pkg/authn/keychain.go:161]: I've never seen actions SO mad at one line before in my life 😂

**Comment by imjasonh** [comment]
Closing in favor of hopefully making ecr-login's logging more configurable: https://github.com/awslabs/amazon-ecr-credential-helper/pull/309

**Pull request closed without merging**